### PR TITLE
docs(pdk) add more docs for the new log serializer function

### PR DESCRIPTION
**Pull request by dndx**

### CODE REVIEW

**Comment by dndx** [comment]
@gszr We have a few other commits on the tip of `next` that need to be cherry-picked into the 2.1 release branch. Maybe it is better to keep them together so that we keep track of which ones need to be cherry-picked.

But if you still believe PR against the release branch is better then I don't have strong feeling about it either :)

**Comment by dndx** [comment]
NVM, I will rebase this PR against the release branch for now, we can cherry pick the `next` commits later.

**Comment by gszr** [comment]
> NVM, I will rebase this PR against the release branch for now, we can cherry pick the `next` commits later.
> 
> One of the commit came from `next`: [612c365](https://github.com/Kong/kong/commit/612c365e5027695318ba40cf436754452879a28b).

There's no need to cherry-pick them into `next`, as they will eventually arrive in `next` (once 2.1 is out, `release/2.1` is merged into `master`, then `master` is merged into `next`, as we regularly do).

**Comment by dndx** [comment]
@gszr Yeah I realized that. Although we do need to cherry pick the above commit because the second commit was based on it.

File: kong/pdk/log.lua [hunk 1]
```diff
@@ -473,10 +473,47 @@ do
...
+  -- The following fields are only present in a TLS/HTTPS request:
+  -- * `request.tls.version` - TLS/SSL version used by the connection.
+  -- * `request.tls.cipher` - TLS/SSL cipher used by the connection.
+  -- * `request.tls.client_verify` - mTLS validation result. Contents are the same as described in [$ssl_client_verify](https://nginx.org/en/docs/http/ngx_http_ssl_module.html#var_ssl_client_verify).
+  --
+  -- **Warning:** This function may return sensitive data (e.g., API keys).
+  -- Consider filtering before writing it to unsecured locations.
+  --
+  -- To see what content is present in your setup, enable any of the logging
+  -- plugins (e.g. ,`file-log`) and the output written to the log file is the table
```
Comment (gszr) [inline kong/pdk/log.lua:509]:
```suggestion
  -- plugins (e.g., `file-log`) and the output written to the log file is the table
```

**Pull request merged**